supported_platforms.md

```diff
@@ -0,0 +1,51 @@
+# Microsoft Azure IoT SDKs for .NET
+
+This SDK is tested nightly on a mix of .NET frameworks on both Windows 10 and on Ubuntu 1604. For additional details
+for each tested platform, see the respective sections below. 
+
+## Supported .NET versions
+
+The NuGet packages provide support for the following .NET versions:
+- .NET Standard 2.1
+- .NET Standard 2.0
+- .NET Framework 4.7.2 (IoT Hub SDKs only)
+- .NET Framework 4.5.1 (IoT Hub SDKs only)
+
+This SDK _may_ work with other versions of .NET, but there are no guarantees that they will _always_ work for those 
+versions nor are there guarantees that we will fix bugs that are only present on those versions.
+
+## Windows 10
+
+Note that, while we only directly test on Windows 10, we do support other Windows versions as well.
+
+Nightly test platform details:
+
+.NET versions tested on
+- .NET Framework 4.5.1
+- .NET Framework 4.7.2
+- .NET Core 2.1.18
+- .NET Core 3.1
+
+Default locale: en_US, platform encoding: Cp1252
+
+OS name: "windows server 2016", version: "10.0", arch: "amd64", family: "windows"
```

Comment:
How about we add a link to this in our yaml file as well? That way we can remember to update this doc whenever we update our build agents